src/tiled/objectselectionitem.cpp

```diff
@@ -184,17 +193,35 @@ void MapObjectOutline::paint(QPainter *painter,
         QLineF(mBoundingRect.topRight(), mBoundingRect.bottomRight())
     };
 
-    QPen dashPen(Qt::DashLine);
+    QPen dashPen(Qt::SolidLine);
     dashPen.setCosmetic(true);
-    dashPen.setDashOffset(qMax(qreal(0), x()));
+
+    // draw a solid white line
+    dashPen.setColor(Qt::white);
+    painter->setPen(dashPen);
+    painter->drawLines(horizontal, 2);
+
+    // Draw a black dashed line above above the white line
+    dashPen.setColor(Qt::black);
+    dashPen.setStyle(Qt::DashLine);
+    dashPen.setDashOffset(qMax(qreal(0), x())+offset);
     painter->setPen(dashPen);
     painter->drawLines(horizontal, 2);
 
-    dashPen.setDashOffset(qMax(qreal(0), y()));
+    dashPen.setColor(Qt::white);
+    dashPen.setStyle(Qt::SolidLine);
     painter->setPen(dashPen);
     painter->drawLines(vertical, 2);
```

Comment:
Drawing the white vertical lines together with the white horizontal lines saves you from having to do two pen adjustments.